WebInstall/explainer_cross_domain.md

```diff
@@ -0,0 +1,293 @@
+* Track campaign IDs for marketing campaigns.
+
+## Non-goals
+
+* Install same-domain content (see [Web Install - same-domain explainer](./explainer_same_domain.md)).
+* Change the way the UA currently prompts for installation of a web app.
+* Associate ratings and reviews with the installed app ([see Ratings and Reviews API explainer](https://github.com/MicrosoftEdge/MSEdgeExplainers/blob/main/RatingsAndReviewsPrompt/explainer.md)).
+* Process payments for installation of PWAs ([see Payment Request API](https://developer.mozilla.org/en-US/docs/Web/API/Payment_Request_API)).
+* List purchased/installed goods from a store ([see Digital Goods API](https://github.com/WICG/digital-goods/blob/main/explainer.md)).
+* Installing content that does not pass the installability criteria (see *[installability criteria](#installability-criteria)*).
+
+## Use Cases
+
+There are two main use cases that the Web Install API enables for cross-domain origins:
+
+### **Web app installation from associated domain**
+
+An (out-of-scope) domain associated to a web app could prompt for the installation of said web app. The typical use case for this is a marketing website that can offer their customers a direct way to install the associated web app.
+
+As an example, a user can navigate to the `streamflix.com` website and find UX that allows them to install the associated application found in `streamflix.app`.
+
+```javascript
+/* Example that uses the Permissions API to check if the installation permission is set before calling the install method */
+  const installApp = async () => {
+    try{
+        const { state } = await navigator.permissions.query({
+            name: "installation"
+          });
+          switch (state) {
+            case "granted":
+                const value = await navigator.install('https://streamflix.app');
+              break;
+            case "prompt":
+              showInstallButton();
+              break;
+            case "denied":
+              browseToAppStorePage();
+              break;
+          }
+    }
+    catch(err){console.error(err.message)}
+  };
+```
+
+Manifest file for the `streamflix.app`, allowing installation *ONLY* from `streamflix.com` :
+
+```json
+{
+    "name": "Streamflix App",
+    "display": "standalone",
+    "start_url": "/index.html",
+    "install_sources": [ 
+	    {"origin": "streamflix.com"}   
+    ]
+}
+```
+
+![Different domain install flow](./difdomaininstall.png) 
+
+### **Creation of online catalogs**
+
+ The other use case for the API is related to the creation of online catalogs. A web site/app can list and install web apps. For example, `store.com` would be able to distribute apps on multiple platforms and multiple devices.
+
+```javascript
+/* tries to install a cross-domain web app */
+
+const installApp = async (manifest_id) => {
+    try {
+        const value = await navigator.install(manifest_id);
```

Comment:
The manifest id doesn't necessarily resolve into a serve-able URL. So it cannot be the only thing populated here.

Can you please require the id and a source to get the app, perhaps `navigator.install(manifest_id, {url: install_url});`

Reply:
I was picturing the argument to be either a manifest_id or a url (for installing content that might not be a web app). but that does promote ambiguity around the id/start_url/url and is not in our best interest. I do like however the idea of including the url as an optional param. 

How do you see this working for installations of content that are not apps? 

`navigator.install('https://streamflix.app', {url:'https://streamflix.app'})`;

`navigator.install('', {url:''https://streamflix.app})`;

`navigator.install({url:'https://streamflix.app'})`; <- guessing this would be the adopted way for non manifest_id content? It might be weird as the omission of the first param is how the same-domain install works.